### PR TITLE
[Issue: 169] Frontend hardcodes python3

### LINKED ISSUE #169: Frontend hardcodes python3
Need to make the decision about what language to use server-side after loading the notebook. That, or the server has encode the kernel name in the page template and let the JS echo it back. 

I don't know which one is better / worse yet. 💭

### PR DESCRIPTION
**Pull request by jhpedemonte**
Fixes #169

### CODE REVIEW

File: Dockerfile.kernel [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 # install Declarative Widgets python package
 # don't bother activating the extension, not needed outside notebook
-RUN pip install jupyter_declarativewidgets
+RUN pip install jupyter_declarativewidgets==0.5.2
+
+# also install above packages for python2
+USER root
+RUN /opt/conda/envs/python2/bin/pip install 'ipywidgets==5.1.3'
+RUN /opt/conda/envs/python2/bin/pip install jupyter_declarativewidgets==0.5.2
+USER jovyan
```
Comment (jhpedemonte) [inline Dockerfile.kernel]: @parente Wouldn't install properly unless I was root. Is this the best way?

**Comment by jhpedemonte** [comment]
OK, I think this is good to go. Runs scala test notebook (as well as all others) and passes integration tests.

File: Dockerfile.kernel (hunk 1 above)
Comment (parente) [inline Dockerfile.kernel]: No. Something's wrong if we can't run those as jovyan. I bet we need to activate the conda env properly. Like `bash -c "source activate python2 && <all the rest of the command lumped in>"`

**Comment by parente** [comment]
Testing this AM.

File: routes/api.js [hunk 2]
```diff
@@ -139,42 +133,62 @@ router.post('/kernels', bodyParser.json({ type: 'text/plain' }), function(req, r
...
+        // Store notebook path for later use
+        sessions[sessionId] = notebookPath;
+
+        nbstore.get(notebookPath)
+        .then(function success(notebook){
+            //debug(notebook.metadata);
+            if (notebook.metadata.kernelspec.name) {
+                var kernelName = notebook.metadata.kernelspec.name;
+                debug("Notebook kernel name found: " + kernelName);
+                if(kernelName === "apache_toree") {
```
Comment (parente) [inline routes/api.js]: Why this mapping? If the kernelName in the notebook is "apache_toree", that should be how the kernel is requested from the KG as well. The notebook server definitely doesn't have any kind of special logic like this so we shouldn't either.

File: routes/api.js [hunk 3]
```diff
@@ -139,42 +133,62 @@ router.post('/kernels', bodyParser.json({ type: 'text/plain' }), function(req, r
...
+                    kernelName = "scala";
+                }
+                req.body.name = kernelName;
+            } else {
+                // Default to Python 3
+                debug("Notebook kernel name not found, defaulting to Python 3");
+                req.body.name = "python3";
```
Comment (parente) [inline routes/api.js]: Don't do this. The KG will be configured to pick the correct default kernel given a blank kernel name.

File: routes/api.js (hunk 2 above)
Comment (parente) [inline routes/api.js]: I see why. The `FROM` line in the `Dockerfile.kernel` is pointing to a build so old, that it predates the Toree kernel and is still using the ET-Spark kernel. Yikes!

So we definitely don't want the translation from apache_toree to scala here. It's making us compatible with the old ET kernel and not Apache Toree.

I suggest moving to 2d878db5cbff which is the last build to have notebook 4.1 and Toree (https://github.com/jupyter/docker-stacks/wiki/Docker-build-history).

File: etc/notebooks/test/test_scala_hello_world.ipynb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      "layout": {
+       "col": 0,
+       "height": 4,
+       "row": 0,
+       "width": 4
+      }
+     }
+    }
+   },
+   "outputs": [
```
Comment (parente) [inline etc/notebooks/test/test_scala_hello_world.ipynb]: Put a markdown cell somewhere in the dashboard. The hello world took so long to appear, I thought my dashboard server was busted. At least the markdown will appear instantly and give some sign of life. (Odd that the spinner bar didn't appear either ...)

**Comment by jhpedemonte** [comment]
Update, @parente, please review.

**Comment by parente** [comment]
LGTM. Thanks for finishing it off!

**Pull request merged**